form_helper - Allow arrays as default and return values in set_value method

najdanovicivan opened the pull request:
**Description**
When array type form fields are used like multiple select it is needed to pass back multiple values as array

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

MGatner reviewed (approved):
Looks good to me. I will leave it in case anyone else wants to set eyes on it. A test case for array input would be awesome, but our `set_value()` tests are already a little on the weak side.

paulbalandan commented on system/Helpers/form_helper.php:
```diff
@@ -713,13 +713,13 @@ function form_close(string $extra = ''): string
 	 * @return string
 	 */
-	function set_value(string $field, string $default = '', bool $htmlEscape = true): string
+	function set_value(string $field, $default = '', bool $htmlEscape = true)
```
Why remove the `: string` return typehint?

najdanovicivan replied:
Becasue the method return value can be string array. Having type hint in such case will throw error.

I see that I missed to update the return type in phpdoc. It should be string|array

paulbalandan replied:
That is what I'm thinking earlier. Good to see the changes.

najdanovicivan commented:
I've updated the code to be more clear. Forms in HTML can return string or string[] so instead of array I've updated the PhpDoc to use string|string[] for both default parameter and return type
